tests/console/docker_image.pm

```diff
@@ -58,30 +58,32 @@ sub run {
         chmod +x container-diff-linux-amd64 && sudo mv container-diff-linux-amd64 /usr/local/bin/container-diff"
     );
 
-    foreach my $image_name (@image_names) {
+    for my $i (0 .. $#image_names) {
         # Load the image
-        assert_script_run("docker pull $image_name", 120);
+        assert_script_run("docker pull $image_names[$i]", 120);
         # Running executables works
-        assert_script_run qq{docker container run --entrypoint '/bin/bash' --rm $image_name -c 'echo "I work" | grep "I work"'};
+        assert_script_run qq{docker container run --entrypoint '/bin/bash' --rm $image_names[$i] -c 'echo "I work" | grep "I work"'};
         # It is the right SLE version
         if (is_sle("=12-SP3")) {
-            validate_script_output("docker container run --entrypoint '/bin/bash' --rm $image_name -c 'cat /etc/os-release'", sub { /PRETTY_NAME="SUSE Linux Enterprise Server 12 SP3"/ });
+            validate_script_output("docker container run --entrypoint '/bin/bash' --rm $image_names[$i] -c 'cat /etc/os-release'", sub { /PRETTY_NAME="SUSE Linux Enterprise Server 12 SP3"/ });
         }
         elsif (is_sle("=15")) {
-            validate_script_output("docker container run --entrypoint '/bin/bash' --rm $image_name -c 'cat /etc/os-release'", sub { /PRETTY_NAME="SUSE Linux Enterprise Server 15"/ });
+            validate_script_output("docker container run --entrypoint '/bin/bash' --rm $image_names[$i] -c 'cat /etc/os-release'", sub { /PRETTY_NAME="SUSE Linux Enterprise Server 15"/ });
         }
         # zypper lr
-        assert_script_run("docker container run --entrypoint '/bin/bash' --rm $image_name -c 'zypper lr -s'", 120);
+        assert_script_run("docker container run --entrypoint '/bin/bash' --rm $image_names[$i] -c 'zypper lr -s'", 120);
         # zypper ref
-        assert_script_run("docker container run --entrypoint '/bin/bash' --rm $image_name -c 'zypper -v ref | grep \"All repositories have been refreshed\"'", 120);
+        assert_script_run("docker container run --entrypoint '/bin/bash' --rm $image_names[$i] -c 'zypper -v ref | grep \"All repositories have been refreshed\"'", 120);
 
         # container-diff
-        my $image_file = $image_name =~ s/\/|:/-/gr;
-        assert_script_run("container-diff analyze daemon://$image_name --type=rpm > /tmp/container-diff-$image_file.txt");
+        my $image_file = $image_names[$i] =~ s/\/|:/-/gr;
+        assert_script_run("docker pull $image_names[$i]");
+        assert_script_run("docker pull $stable_names[$i]");
+        assert_script_run("container-diff diff daemon://$image_names[$i] daemon://$stable_names[$i] --type=rpm --type=file --type=history > /tmp/container-diff-$image_file.txt");
         upload_asset("/tmp/container-diff-$image_file.txt");
```

Comment:
I just noticed this, should be this diff uploaded as asset ? I guess it should be done just with `upload_logs`